code/datums/martial/rising_bass.dm

```diff
@@ -0,0 +1,163 @@
+#define SIDE_KICK_COMBO "DH"
+#define SHOULDER_FLIP_COMBO "GHDGGHD"
+#define REPULSE_PUNCH_COMBO "GHGH"
+#define FOOT_SMASH_COMBO "HH"
+#define DEFT_SWITCH_COMBO "GDD"
+
+/datum/martial_art/the_rising_bass
+	name = "The Rising Bass"
+	id = MARTIALART_RISINGBASS
+	dodge_chance = 100
+	no_guns = TRUE
+	allow_temp_override = FALSE
+	help_verb = /mob/living/carbon/human/proc/rising_bass_help
+
+/datum/martial_art/the_rising_bass/proc/check_streak(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(findtext(streak,SIDE_KICK_COMBO))
+		streak = ""
+		sideKick(A,D)
+		return 1
+	if(findtext(streak,SHOULDER_FLIP_COMBO))
+		streak = ""
+		shoulderFlip(A,D)
+		return 1
+	if(findtext(streak,REPULSE_PUNCH_COMBO))
+		streak = ""
+		repulsePunch(A,D)
+		return 1
+	if(findtext(streak,FOOT_SMASH_COMBO))
+		streak = ""
+		footSmash(A,D)
+		return 1
+	if(findtext(streak,DEFT_SWITCH_COMBO))
+		streak = ""
+		deftSwitch(A,D)
+		return 1
+	return 0
+
+
+/datum/martial_art/the_rising_bass/proc/sideKick(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!D.IsKnockdown())
+		var/turf/H
+		switch(A.dir)
+			if(NORTH)
+				H = get_step(get_step(A,NORTH),pick(EAST,WEST))
+			if(EAST)
+				H = get_step(get_step(A,EAST),pick(SOUTH,NORTH))
+			if(SOUTH)
+				H = get_step(get_step(A,SOUTH),pick(EAST,WEST))
+			if(WEST)
+				H = get_step(get_step(A,WEST),pick(SOUTH,NORTH))
+		A.do_attack_animation(D, ATTACK_EFFECT_KICK)
+		D.visible_message("<span class='warning'>[A] kicks [D] in the side, sliding them over!</span>", \
+						  "<span class='userdanger'>[A] kicks you in the side, forcing you to step away!</span>")
+		playsound(get_turf(A), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+		D.apply_damage(5, BRUTE, BODY_ZONE_CHEST)
+		D.Knockdown(60)
+		var/turf/L = H
+		for(var/obj/i in H.contents)
+			if(!istype(i,/mob) && i.density == 1)//(i.anchored == 1 && i.density == 1) || istype(i,/obj/structure) || istype(i,/turf/closed)
+				L = D.loc
+		D.forceMove(L)
+		return 1
+	log_combat(A, D, "side kicked (Rising Bass)")
+	return basic_hit(A,D)
+
+/datum/martial_art/the_rising_bass/proc/shoulderFlip(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!D.IsKnockdown())
```

Comment:
This wouldn't work thanks to how bad citadel "stun-combat" is codewise.
Take a look at the krav maga's leg sweep for how I had to work around this issue.